src/ngraph/runtime/cpu/builder/quantization.cpp

```diff
@@ -165,24 +166,58 @@ namespace ngraph
 
                     auto scale_const_op =
                         std::dynamic_pointer_cast<ngraph::op::Constant>(quantize->get_argument(1));
+                    std::vector<float> scales;
                     if (scale_const_op == nullptr)
                     {
-                        throw ngraph_error("Quantize scale must be a constant");
-                    }
-                    auto scale = scale_const_op->get_vector<float>();
+                        auto& arg1_tensor = external_function->get_tensor_data(args[1].get_name());
+                        auto scales_size = shape_size(args[1].get_shape());
 
-                    std::vector<float> scales;
-                    scales.push_back(1.0 / scale[0]);
-                    size_t quantize_index =
-                        mkldnn_emitter->build_quantize_reorder(input_desc, result_desc, scales);
-                    auto& deps = mkldnn_emitter->get_primitive_deps(quantize_index);
-                    auto functor = [&, quantize_index](CPURuntimeContext* ctx,
-                                                       CPUExecutionContext* ectx) {
-                        cpu::mkldnn_utils::set_memory_ptr(ctx, deps[0], arg0_tensor);
-                        cpu::mkldnn_utils::set_memory_ptr(ctx, deps[1], out_tensor);
-                        cpu::mkldnn_utils::mkldnn_invoke_primitive(ctx, quantize_index);
-                    };
-                    functors.emplace_back(functor);
+                        // Dummy value while we wait for the actual values that are provided during
+                        // execution
+                        scales.push_back(1.0f);
+                        size_t quantize_index =
+                            mkldnn_emitter->build_quantize_reorder(input_desc, result_desc, scales);
+                        auto& deps = mkldnn_emitter->get_primitive_deps(quantize_index);
+                        auto functor = [&, input_desc, result_desc, scales_size](
+                            CPURuntimeContext* ctx, CPUExecutionContext* ectx) {
+                            // Create MKLDNN reorder primitive during the first iteration.
+                            // Assumes the scales dont change for the duration of the graph
+                            static mkldnn::reorder r_prim = [&]() {
```

Comment:
can we make sure this static gets recreated for a different instance of quantize op.

Reply:
It shouldn't matter -- if it references something in the functor and/or node, it will be recreated, if not, it doesn't matter.

Reply:
@jbobba i think scott addressed it, i wasnt sure if this static variable in this lambda behaves just like regular function static variable (where it gets instantiated once per func definition). for eg:
https://stackoverflow.com/questions/16750933/static-variables-in-c-lambdas